pkg/api/payer/service.go

```diff
@@ -489,53 +517,92 @@ func (s *Service) signClientEnvelope(originatorID uint32,
 		},
 		TargetOriginator: originatorID,
 		MessageRetentionDays: determineRetentionPolicy(
-			clientEnvelope.TargetTopic(),
-			clientEnvelope.Aad(),
+			clientEnvelope.payload.TargetTopic(),
+			clientEnvelope.isCommit,
 		),
 	}, nil
 }
 
 func determineRetentionPolicy(
 	targetTopic topic.Topic,
-	aad *envelopesProto.AuthenticatedData,
+	isCommit *bool,
 ) uint32 {
 	// TODO: mkysel determine expiration for welcomes and key packages
 
 	switch targetTopic.Kind() {
 	case topic.TOPIC_KIND_IDENTITY_UPDATES_V1:
 		panic("should not be called for identity updates")
 	case topic.TOPIC_KIND_GROUP_MESSAGES_V1:
-		if aad.GetIsCommit() {
+		if isCommit == nil {
+			panic("isCommit should not be nil for group messages")
+		}
+		if *isCommit {
 			panic("should not be called for commits")
 		}
 	}
 
 	return constants.DEFAULT_STORAGE_DURATION_DAYS
 }
 
-func shouldSendToBlockchain(targetTopic topic.Topic, aad *envelopesProto.AuthenticatedData) bool {
-	switch targetTopic.Kind() {
+// shouldSendToBlockchain determines whether a given ClientEnvelope should be forwarded to the blockchain.
+//
+// It inspects the topic kind of the envelope and, for group messages, invokes validation logic to determine whether the
+// message represents a commit.
+//
+// Parameters:
+//   - clientEnvelope (*envelopes.ClientEnvelope): The client envelope to inspect and validate.
+//
+// Returns:
+//
+//   - bool: A general indication of whether the message should be sent to the blockchain.
+//
+//   - true: send to blockchain
+//
+//   - false: do not send
+//
+//   - *bool: A specific indication of commit status (only for group messages):
+//
+//   - nil: not a group message or unknown commit state
+//
+//   - non-nil pointer to bool:
+//
+//   - true: this is a commit and should be sent
+//
+//   - false: this is not a commit and should not be sent
+//
+//   - error: Any error encountered during validation. If non-nil, the decision is considered invalid.
+func (s *Service) shouldSendToBlockchain(
```

Comment:
not the most elegant solution, but I dont want to introduce a large refactor. Maybe in the future.